library/src/main/java/com/vinted/coper/CoperImpl.kt

```diff
@@ -59,6 +72,52 @@ internal class CoperImpl(private val fragmentManager: FragmentManager) : Coper {
         }
     }
 
+    internal suspend fun getFragmentSafely(): CoperFragment {
+        val deferred = CompletableDeferred<CoperFragment>()
+        val fragment = getAttachedFragment()
+        fragment.lifecycle.addObserver(object : DefaultLifecycleObserver {
+            override fun onCreate(owner: LifecycleOwner) {
+                deferred.complete(fragment)
+            }
+        })
+        return deferred.await()
+    }
+
+    private suspend fun getAttachedFragment(): CoperFragment {
+        return fragmentInitializationMutex.withLock {
+            suspendCancellableCoroutine { continuation ->
+                val fragment = fragmentManager
+                    .findFragmentByTag(FRAGMENT_TAG) as? CoperFragment
+                if (fragment == null) {
+                    val newFragment = CoperFragment()
+                    if (lifecycle != null) {
+                        lifecycle.addObserver(object : DefaultLifecycleObserver {
+                            override fun onCreate(owner: LifecycleOwner) {
+                                attachFragment(newFragment, continuation::resume)
+                            }
+                        })
```

Comment:
Are you sure it is safe? What if lifecycle already passed `CREATED` state? Maybe you should use `lifecycle.currentState.isAtLeast` api?

Reply:
```
     * The given observer will be brought to the current state of the LifecycleOwner.
     * For example, if the LifecycleOwner is in {@link State#STARTED} state, the given observer
     * will receive {@link Event#ON_CREATE}, {@link Event#ON_START} events.
```

Reply:
As I understand it will give all events

Reply:
Do you have a test that proves that?

Reply:
good catch, I will do this in a moment